### PR TITLE
Feature test scenario

### PR DESCRIPTION
**Pull request by RomaSliva**
add some methods for parameters, test scenarios, actScenarios and deleted useless one

### CODE REVIEW

File: src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java [hunk 1]
```diff
@@ -32,18 +32,31 @@ public PaginationActScenario findByTitlePaginated(@RequestParam("title") String
                                                  @RequestParam("page") int page){
         return actScenarioService.findByTitleLike(title,size,page);
     }
+
+    //same as without /all but doesn`t have pagination
+    @GetMapping("/test-scenarios/act-scenarios/all")
+    public List<ActScenario> findAllByTitle(@RequestParam("title") String title){
+        return actScenarioService.findAllByTitleLike(title);
+    }
+
```
Comment (henchamax) [inline src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java:41]: Do you really need get all operation ?

File: src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java [hunk 2]
```diff
@@ -32,18 +32,31 @@ public PaginationActScenario findByTitlePaginated(@RequestParam("title") String
...
-    @GetMapping("/test-scenarios/act-scenarios/all")
-    public List<ActScenario> findAllByTitlePaginated(@RequestParam("title") String title){
-        return actScenarioService.findAllByTitleLike(title);
+
+    @GetMapping("/test-scenarios/{idTestScenario}/gparams")
+    public List<ActScenario> findByIdInTestScen(@PathVariable int idTestScenario){
+        return actScenarioService.findByIdInTestScenario(idTestScenario);
     }
+
+    @GetMapping("/test-scenarios/{idTestScenario}/g")
```
Comment (henchamax) [inline src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java]: What does "g" prefix means ?

File: src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java [hunk 3]
```diff
@@ -53,6 +66,10 @@ public PaginationActScenario findByTitleInTestScenPaginated(@RequestParam("title
     public ActScenario editActScenario(@RequestBody ActScenario actScenario, @PathVariable int idActScen){
         return actScenarioService.editActScenario(actScenario,idActScen);
     }
+    @PutMapping("/act-scenarios/s/{idActScen}")
```
Comment (henchamax) [inline src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java]: What does "s" means ?

File: src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java [hunk 4]
```diff
@@ -79,9 +90,9 @@ public void deleteTestScenario(int id) {
     }
 
     @Override
-    public int getTotalElements(int idTestCase, String title) {
+    public int getTotalElementsByTitleAndIdTestCase(int idTestCase, String title) {
```
Comment (henchamax) [inline src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java:93]: Title is case sensitive ?

File: src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java [hunk 5]
```diff
@@ -35,6 +34,18 @@ public TestScenarioDaoImpl(JdbcTemplate jdbcTemplate) {
                 new TestScenarioMapper());
     }
 
+    @Override
+    public List<TestScenario> findByTitleLike(String title, int limit, int offset) {
+        return jdbcTemplate.query(
+                FIND_BY_TITLE_LIKE,
```
Comment (henchamax) [inline src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java:40]: Title is case sensitive ?

File: src/main/java/com/ncedu/cheetahtest/service/testscenario/TestScenarioServiceImpl.java [hunk 6]
```diff
@@ -60,46 +72,37 @@ public TestScenario createTestScenario(TestScenario testScenario, List<ActionsAn
...
         testScenario.setStatus(ACTIVE);
+        TestScenario createdTestScen;
         createdTestScen = testScenarioDao.createTestScenario(testScenario);
 
         ActScenario actScenario = new ActScenario();
 
         int priority = 1;
-        for (ActionsAndCompoundsID actAndComp: actAndCompID) {
+        for (ActionsAndCompoundsID actAndComp : actAndCompID) {
             if (actAndComp.getKind().equals("ACTION")){
```
Comment (henchamax) [inline src/main/java/com/ncedu/cheetahtest/service/testscenario/TestScenarioServiceImpl.java:82]: Will be better to use enum for actAndComp kind

File: src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java (hunk 5 above)
Comment (RomaSliva) [reply src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java:40]: no

File: src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java (hunk 4 above)
Comment (RomaSliva) [reply src/main/java/com/ncedu/cheetahtest/dao/testscenario/TestScenarioDaoImpl.java:93]: no

File: src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java (hunk 3 above)
Comment (RomaSliva) [reply src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java]: renamed to set

File: src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java (hunk 2 above)
Comment (RomaSliva) [reply src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java]: renamed to general, it will be more clear

**Review by GorovDimon11 — APPROVED** [review]

File: src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java (hunk 1 above)
Comment (RomaSliva) [reply src/main/java/com/ncedu/cheetahtest/controller/actscenario/ActScenarioController.java:41]: Not yet, if we will not use it, we delete.

File: src/main/java/com/ncedu/cheetahtest/service/testscenario/TestScenarioServiceImpl.java (hunk 6 above)
Comment (RomaSliva) [reply src/main/java/com/ncedu/cheetahtest/service/testscenario/TestScenarioServiceImpl.java:82]: it is safe anyway

**Review by danilos1 — APPROVED** [review]

**Pull request merged**